src/js/client/login.js

```diff
@@ -0,0 +1,103 @@
+// provide a front-end to /api/user/login
+/* global fluid, jQuery */
+(function () {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.express.user.frontend.login");
+
+        // If the user controls are used to log out, we have to manually clear the success message.
+    // If we delegate this to the controls component, it might clobber success messages for things other than the login.
+    gpii.express.user.frontend.login.checkAndClearSuccess = function (that) {
+        if (!that.model.user || !that.model.user.username) {
+            that.applier.change("successMessage", null);
+            that.renderInitialMarkup();
+        }
+    };
+
+    fluid.defaults("gpii.express.user.frontend.login", {
+        gradeNames: ["gpii.express.user.frontend.canHandleStrings", "gpii.templates.templateFormControl"],
+        templates: {
+            initial: "login-viewport",
+            error:   "common-error",
+            success: "common-success"
+        },
+        model: {
+            user: null
+        },
+        components: {
+            success: {
+                options: {
+                    model: {
+                        message: "{login}.model.successMessage"
+                    },
+                    modelListeners: {
+                        // TODO:  Review with Antranig to confirm why the rules in `templateMessage` aren't enough to handle this.
```

Comment:
Not clear what this comment is referring to - perhaps the base grade "templateFormControl"? I don't see that it has any modelListeners defined in it.